src/translate-adapter.ts

```diff
@@ -0,0 +1,455 @@
+import { gray, red, yellow } from "ansi-colors";
+import {
+	ensureDir,
+	existsSync,
+	readFile,
+	readJson,
+	stat,
+	writeFile,
+	writeJson,
+} from "fs-extra";
+import { glob } from "glob";
+import { EOL } from "os";
+import { promisify } from "util";
+import { translateText } from "./translate";
+import { escapeRegExp, padRight } from "./util";
+import path = require("path");
+import yargs = require("yargs/yargs");
+
+let ioPackage: string;
+let admin: string;
+let words: string;
+let i18nBases: string[];
+
+const parser = yargs(process.argv.slice(2));
+parser
+	.env("IOBROKER_TRANSLATE")
+	.strict()
+	.usage("ioBroker adapter translator\n\nUsage: $0 <command> [options]")
+	.alias("h", "help")
+	.alias("v", "version")
+	.command(
+		["translate", "t"],
+		"Translate io-package.json and all admin language files",
+		{},
+		interceptErrors(handleTranslateCommand),
+	)
+	.command(
+		["to-json", "adminWords2languages", "j"],
+		"Convert words.js to i18n JSON files",
+		{},
+		interceptErrors(handleToJsonCommand),
+	)
+	.command(
+		["to-words", "adminLanguages2words", "w"],
+		"Generate words.js from i18n JSON files",
+		{},
+		interceptErrors(handleToWordsCommand),
+	)
+	.command(
+		["all", "translateAndUpdateWordsJS", "a"],
+		"Sequence of translate, to-words, to-json",
+		{},
+		interceptErrors(handleAllCommand),
+	)
+	/*
+	translateAndUpdateWordsJS: TaskFunction;*/
+	.options({
+		ioPackage: {
+			type: "string",
+			alias: "p",
+			default: "./io-package.json",
+			description: "Path to the io-package.json file",
+		},
+		admin: {
+			type: "string",
+			alias: "a",
+			default: "./admin",
+			description: "Path to the admin directory",
+		},
+		words: {
+			type: "string",
+			alias: "w",
+			description: "Path to the words.js file",
+		},
+		base: {
+			type: "string",
+			alias: "b",
+			array: true,
+			description:
+				"Path to the english i18n file, multiple files are possible",
+		},
+	})
+	.middleware(setDirectories)
+	.wrap(Math.min(100, parser.terminalWidth()))
+	.help().argv;
+
+/********************************** Helpers ***********************************/
+
+const _languages = {
```

Comment:
If you want to strongly type this, make it a `Record<ioBroker.Languages, /* whatever */>`.
We have a central type for that, no need to re-define.